easybuild/easyblocks/a/amber.py

```diff
@@ -0,0 +1,189 @@
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://vscentrum.be/nl/en),
+# the Hercules foundation (http://www.herculesstichting.be/in_English)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# http://github.com/hpcugent/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for Amber, implemented as an easyblock
+
+@author: Benjamin Roberts (The University of Auckland)
+@author: Kenneth Hoste (Ghent University)
+"""
+import os
+
+from easybuild.easyblocks.generic.configuremake import ConfigureMake
+from easybuild.framework.easyconfig import CUSTOM, MANDATORY, BUILD
+from easybuild.tools.build_log import EasyBuildError
+import easybuild.tools.environment as env
+from easybuild.tools.run import run_cmd
+from easybuild.tools.modules import get_software_root
+import easybuild.tools.toolchain as toolchain
+
+class EB_Amber(ConfigureMake):
+    """Easyblock for building and installing Amber"""
+
+    @staticmethod
+    def extra_options(extra_vars=None):
+        """Extra easyconfig parameters specific to ConfigureMake."""
+        extra_vars = dict(ConfigureMake.extra_options(extra_vars))
+        extra_vars.update({
+            # 'Amber': [True, "Build Amber in addition to AmberTools", CUSTOM],
+            'patchlevels': ["latest", "(AmberTools, Amber) updates to be applied", CUSTOM],
+        })
+        return ConfigureMake.extra_options(extra_vars)
+
+    def __init__(self, *args, **kwargs):
+        super(EB_Amber, self).__init__(*args, **kwargs)
+        self.already_extracted = False
+        self.build_in_installdir = True
+
+    def extract_step(self):
+        """Only extract from the tarball if this has not already been done."""
+        if (self.already_extracted == True):
+            pass
+        else:
+            # unpack_options is a string, not an array; can't use append
+            self.cfg['unpack_options'] += " --strip-components=1"
+            super(EB_Amber, self).extract_step()
+            self.already_extracted = True
+
+    def patch_step(self, **kw):
+        env.setvar('AMBERHOME', self.installdir)
+        if self.cfg['patchlevels'] == "latest":
+            cmd = "./update_amber --update"
+            # This needs to be run multiple times, more's the pity.
```

Comment:
Is twice gonna suffice? Else I would suggest that you print a warning about this.